doesn't work wo shards in relay mode

```
panic: no shard configured
goroutine 1 [running]:
github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine.IterateShards(0xc00006ff00, 0xc0002c1b38)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/config.go:28 +0x569
main.initShardOptions(0xc000246000)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/config.go:359 +0x77
main.initLocalStorage(0xc000246000)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/config.go:319 +0x45
main.initCfg(0x7ffd7da0feef, 0x1e, 0x2)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/config.go:309 +0xe8a
main.main()
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/main.go:52 +0x14f
```

--- a/cmd/neofs-node/config.go
+++ b/cmd/neofs-node/config.go
@@ -367,7 +367,9 @@ func initLocalStorage(c *cfg) {
 func initShardOptions(c *cfg) {
 	var opts [][]shard.Option
 
-	engineconfig.IterateShards(c.appCfg, func(sc *shardconfig.Config) {
+	require := !nodeconfig.Relay(c.appCfg) // relay node does not require shards
+
+	engineconfig.IterateShards(c.appCfg, require, func(sc *shardconfig.Config) {
 		var writeCacheOpts []writecache.Option
 
 		useWriteCache := sc.UseWriteCache()

--- a/cmd/neofs-node/config/engine/config.go
+++ b/cmd/neofs-node/config/engine/config.go
@@ -19,13 +19,17 @@ const (
 // of "shard" subsection of "storage" section of c, wrap them into
 // shardconfig.Config and passes to f.
 //
-// Panics if N is not a positive number.
-func IterateShards(c *config.Config, f func(*shardconfig.Config)) {
+// Panics if N is not a positive number while shards are required.
+func IterateShards(c *config.Config, required bool, f func(*shardconfig.Config)) {
 	c = c.Sub(subsection)
 
 	num := config.Uint(c, "shard_num")
 	if num == 0 {
-		panic("no shard configured")
+		if required {
+			panic("no shard configured")
+		}
+
+		return
 	}
 
 	def := c.Sub("default")

--- a/cmd/neofs-node/config/engine/config_test.go
+++ b/cmd/neofs-node/config/engine/config_test.go
@@ -17,8 +17,18 @@ func TestEngineSection(t *testing.T) {
 		empty := configtest.EmptyConfig()
 
 		require.Panics(t, func() {
-			engineconfig.IterateShards(empty, nil)
+			engineconfig.IterateShards(empty, true, nil)
 		})
+
+		handlerCalled := false
+
+		require.NotPanics(t, func() {
+			engineconfig.IterateShards(empty, false, func(_ *shardconfig.Config) {
+				handlerCalled = true
+			})
+		})
+
+		require.False(t, handlerCalled)
 
 		require.EqualValues(t, engineconfig.ShardPoolSizeDefault, engineconfig.ShardPoolSize(empty))
 	})
@@ -30,7 +40,7 @@ func TestEngineSection(t *testing.T) {
 
 		require.EqualValues(t, 15, engineconfig.ShardPoolSize(c))
 
-		engineconfig.IterateShards(c, func(sc *shardconfig.Config) {
+		engineconfig.IterateShards(c, true, func(sc *shardconfig.Config) {
 			defer func() {
 				num++
 			}()

--- a/config/example/node.yaml
+++ b/config/example/node.yaml
@@ -84,6 +84,7 @@ object:
     pool_size_remote: 100  # number of async workers for remote PUT operations
 
 storage:
+  # note: shard configuration can be omitted for relay node (see `node.relay`)
   shard_pool_size: 15 # size of per-shard worker pools used for PUT operations
   shard_num: 2  # total number of shards
   default: # section with the default shard parameters

